src/linux/component.rs

```diff
@@ -1,160 +1,389 @@
+            "input" => {
+                let temperature = get_temperature_from_file(&hwmon_file);
+                component.input_file = hwmon_file;
+                component.temperature = temperature;
+                // Max computed value taken form current temperatue if not possible to
+                // get it from `highest`.
+                if let None = component.max {
+                    component.max = temperature;
                 }
             }
+            "label" => component.label = get_file_line(&hwmon_file, 10).unwrap_or(String::new()),
+            "highest" => {
+                component.max = get_temperature_from_file(&hwmon_file);
+                component.highest_file = hwmon_file;
+            }
+            "max" => component.threshold_max = get_temperature_from_file(&hwmon_file),
+            "min" => component.threshold_min = get_temperature_from_file(&hwmon_file),
+            "crit" => component.threshold_critical = get_temperature_from_file(&hwmon_file),
+            // Raspberry_pi weird comportement.
+            "" if component.label.is_empty() => component.label = format!("Component name {}", id),
+            // TODO maybe it's good to trace for being future aware of a change in hwmon temp?
+            _ => {}
+        }
+    };
+
+    let dir = read_dir(folder).ok()?;
+    let mut matchings: HashMap<u32, Component> = HashMap::with_capacity(10);
+    for entry in dir.flatten() {
+        let entry = entry.path();
+        let filename = entry.file_name().and_then(|x| x.to_str()).unwrap_or("");
+        if entry.is_dir() || !filename.starts_with("temp") {
+            continue;
         }
+
+        let (id, item) = filename.split_once('_')?;
+        let id = id.get(4..)?.parse::<u32>().ok()?;
+
+        let component = matchings.entry(id).or_insert_with(|| Component::default());
+        let name = get_file_line(&folder.join("name"), 16);
+        component.name = name.unwrap_or(String::new());
+        let device_model = get_file_line(&folder.join("device/model"), 16);
+        component.device = device_model.unwrap_or(String::new());
+        fill_component(component, id, item, filename);
     }
+    let compo = matchings.into_iter().map(|(k, mut c)| {
+        // sysinfo expose a generic interface with a `label`.
+        // Problem: lot of sensor don't have a label or a device model! ¯\_(ツ)_/¯
+        // So let's pretend we have a unique label!
+        //
+        // All sensor have a `name` on linux hwmon it's mandatory.
+        // - if a sensor has no label: check if don't have a model name, if so it's use it.
+        // - if not do something from id and name at best.
+        if c.label.is_empty() {
+            c.label = if c.device.is_empty() {
+                format!("{} {}", c.name, k)
+            } else {
+                format!("{} {} {}", c.name, c.device, k)
+            }
+        }
+        c
+    });
```

Comment:
After a discussion on XMPP with @linkmauve concluded that adding name anyway is a good idea, would avoid meaningless name like "Composite 1" and gives "nvme Composite 1"

Reply:
by the way thanks for the test on your computer/arm boxes!